Framework/TestWorkflows/src/o2_sim_its_ALP3.cxx

```diff
@@ -92,106 +92,9 @@ DataProcessorSpec sim_its_ALP3() {
-        const double kSensThick = 18e-4;
-        const double kPitchZ = 29.24e-4;
-        const double kPitchX = 26.88e-4;
-        const int    kNRow   = 512;
-        const int    kNCol   = 1024;
-        const double kSiThickIB = 50e-4;
-        const double kSiThickOB = 50e-4;
-        //  const double kSensThick = 120e-4;   // -> sensor Si thickness
-        //
-        const double kReadOutEdge = 0.12;   // width of the readout edge (passive bottom)
-        const double kTopEdge = 37.44e-4;   // dead area on top
-        const double kLeftRightEdge   = 29.12e-4; // width of passive area on left/right of the sensor
-        //
-        const int kNLr = 7;
-        const int kNLrInner = 3;
-        const int kBuildLevel = 0;
-        enum { kRmn, kRmd, kRmx, kNModPerStave, kPhi0, kNStave, kNPar };
-        // Radii are from last TDR (ALICE-TDR-017.pdf Tab. 1.1, rMid is mean value)
-        const double tdr5dat[kNLr][kNPar] = {
-          {2.24, 2.34, 2.67,  9., 16.42, 12}, // for each inner layer: rMin,rMid,rMax,NChip/Stave, phi0, nStaves
-          {3.01, 3.15, 3.46,  9., 12.18, 16},
-          {3.78, 3.93, 4.21,  9.,  9.55, 20},
-          {-1,  19.6 ,   -1,  4.,  0.  , 24},  // for others: -, rMid, -, NMod/HStave, phi0, nStaves // 24 was 49
-          {-1,  24.55, -1,    4.,  0.  , 30},  // 30 was 61
-          {-1,  34.39, -1,    7.,  0.  , 42},  // 42 was 88
-          {-1,  39.34, -1,    7.,  0.  , 48}   // 48 was 100
-        };
-        const int nChipsPerModule = 7; // For OB: how many chips in a row
-        const double zChipGap = 0.01;  // For OB: gap in Z between chips
-        const double zModuleGap = 0.01;// For OB: gap in Z between modules
-
-        // Delete the segmentations from previous runs
-        // FIXME: this should probably be done on each iteration?
-        gSystem->Exec(" rm itsSegmentations.root ");
-
-        // create segmentations:
-        o2::ITSMFT::SegmentationPixel* seg0 = new o2::ITSMFT::SegmentationPixel(
-          0,           // segID (0:9)
-          1,           // chips per module
-          kNCol,       // ncols (total for module)
-          kNRow,       // nrows
-          kPitchX,     // default row pitch in cm
-          kPitchZ,     // default col pitch in cm
-          kSensThick,  // sensor thickness in cm
-          -1,          // no special left col between chips
-          -1,          // no special right col between chips
-          kLeftRightEdge, // left
-          kLeftRightEdge, // right
-          kTopEdge, // top
-          kReadOutEdge // bottom
-          );           // see SegmentationPixel.h for extra options
-        seg0->Store(o2::ITS::GeometryTGeo::getITSSegmentationFileName());
-        seg0->Print();
-
-        double dzLr, rLr, phi0, turbo;
-        int nStaveLr, nModPerStaveLr;
-
-        o2::ITS::Detector* its = new o2::ITS::Detector("ITS", kTRUE, kNLr);
+        o2::ITS::Detector* its = new o2::ITS::Detector("ITS", kTRUE);
```

Comment:
I would propose (for all detectors) to avoid passing the name of the detector via constructor and instead add to all of them a data member (better to put it in the Detectors/Base/Detector.h)

``
using DetID = o2::Base::DetID;
...
DetID mDetID = DetID("ITS"); // or = DetID(DetID::ITS);
``
and assign the base TNamed as SetName(mDetID.getName();
This would assure that the same name is always used

Reply:
yes. good idea.